security: TestTLSCipherRestrict failed

security.TestTLSCipherRestrict [failed](https://mesolite.cluster.engflow.com/invocations/default/e08c4e01-dd23-4de5-9013-053146271c00?testReportRun=1&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc2VjdXJpdHk6c2VjdXJpdHlfdGVzdA==) on master @ [5feef230f17a153f81f2190ffc952c63cc0879d8](https://github.com/cockroachdb/cockroach/commits/5feef230f17a153f81f2190ffc952c63cc0879d8):


```
=== RUN   TestTLSCipherRestrict
    test_log_scope.go:171: test logs captured to: outputs.zip/logTestTLSCipherRestrict1219001882
    test_log_scope.go:82: use -show-logs to present logs inline
    tls_test.go:267: -- test log scope end --
--- FAIL: TestTLSCipherRestrict (3.45s)
=== RUN   TestTLSCipherRestrict/invalid_ciphers
    test_server_shim.go:561: DRPC is enabled (override by TestingGlobalDRPCOption)
    tls_test.go:256: expected wantError=true, got err=<nil>
--- FAIL: TestTLSCipherRestrict/invalid_ciphers (0.59s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/security-engineering @cockroachdb/product-security
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTLSCipherRestrict.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-54606

security.TestTLSCipherRestrict [failed](https://mesolite.cluster.engflow.com/invocations/default/910b69e0-d084-4495-8c1e-642f60e362d0?testReportRun=1&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc2VjdXJpdHk6c2VjdXJpdHlfdGVzdA==) on master @ [a57ed482d55a8fdfa789d6e8e1a87c2de8523f8d](https://github.com/cockroachdb/cockroach/commits/a57ed482d55a8fdfa789d6e8e1a87c2de8523f8d):


```
=== RUN   TestTLSCipherRestrict
    test_log_scope.go:171: test logs captured to: outputs.zip/logTestTLSCipherRestrict1509066970
    test_log_scope.go:82: use -show-logs to present logs inline
    tls_test.go:267: -- test log scope end --
--- FAIL: TestTLSCipherRestrict (3.56s)
=== RUN   TestTLSCipherRestrict/invalid_ciphers
    test_server_shim.go:561: DRPC is enabled (override by TestingGlobalDRPCOption)
    tls_test.go:256: expected wantError=true, got err=<nil>
--- FAIL: TestTLSCipherRestrict/invalid_ciphers (0.76s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTLSCipherRestrict.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

security.TestTLSCipherRestrict [failed](https://mesolite.cluster.engflow.com/invocations/default/40980aa2-428c-4f0b-abad-b5e9d48a158f?testReportRun=1&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc2VjdXJpdHk6c2VjdXJpdHlfdGVzdA==) on master @ [223617d5f65835071f1f3395bc7b487e6dc029c9](https://github.com/cockroachdb/cockroach/commits/223617d5f65835071f1f3395bc7b487e6dc029c9):


```
=== RUN   TestTLSCipherRestrict
    test_log_scope.go:171: test logs captured to: outputs.zip/logTestTLSCipherRestrict2142185332
    test_log_scope.go:82: use -show-logs to present logs inline
    tls_test.go:267: -- test log scope end --
test logs left over in: outputs.zip/logTestTLSCipherRestrict2142185332
--- FAIL: TestTLSCipherRestrict (13.90s)
=== RUN   TestTLSCipherRestrict/invalid_ciphers
    test_server_shim.go:561: DRPC is enabled (override by TestingGlobalDRPCOption)
    tls_test.go:256: expected wantError=true, got err=<nil>
--- FAIL: TestTLSCipherRestrict/invalid_ciphers (3.48s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTLSCipherRestrict.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

security.TestTLSCipherRestrict [failed](https://mesolite.cluster.engflow.com/invocations/default/616438e1-3f9b-43b1-8fd9-d11d102091da?testReportRun=3&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc2VjdXJpdHk6c2VjdXJpdHlfdGVzdA==) on master @ [223617d5f65835071f1f3395bc7b487e6dc029c9](https://github.com/cockroachdb/cockroach/commits/223617d5f65835071f1f3395bc7b487e6dc029c9):


```
=== RUN   TestTLSCipherRestrict
    test_log_scope.go:171: test logs captured to: outputs.zip/logTestTLSCipherRestrict3328546848
    test_log_scope.go:82: use -show-logs to present logs inline
    tls_test.go:267: -- test log scope end --
test logs left over in: outputs.zip/logTestTLSCipherRestrict3328546848
--- FAIL: TestTLSCipherRestrict (3.54s)
=== RUN   TestTLSCipherRestrict/invalid_ciphers
    test_server_shim.go:561: DRPC is enabled (override by TestingGlobalDRPCOption)
    tls_test.go:256: expected wantError=true, got err=<nil>
--- FAIL: TestTLSCipherRestrict/invalid_ciphers (0.53s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=3</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTLSCipherRestrict.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

security.TestTLSCipherRestrict [failed](https://mesolite.cluster.engflow.com/invocations/default/a43cfac3-ff10-4bfa-a36a-8eb096549e42?testReportRun=3&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc2VjdXJpdHk6c2VjdXJpdHlfdGVzdA==) on master @ [223617d5f65835071f1f3395bc7b487e6dc029c9](https://github.com/cockroachdb/cockroach/commits/223617d5f65835071f1f3395bc7b487e6dc029c9):


```
=== RUN   TestTLSCipherRestrict
    test_log_scope.go:171: test logs captured to: outputs.zip/logTestTLSCipherRestrict2403514666
    test_log_scope.go:82: use -show-logs to present logs inline
    tls_test.go:267: -- test log scope end --
test logs left over in: outputs.zip/logTestTLSCipherRestrict2403514666
--- FAIL: TestTLSCipherRestrict (5.98s)
=== RUN   TestTLSCipherRestrict/invalid_ciphers
    test_server_shim.go:561: DRPC is enabled (override by TestingGlobalDRPCOption)
    tls_test.go:256: expected wantError=true, got err=<nil>
--- FAIL: TestTLSCipherRestrict/invalid_ciphers (1.57s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=3</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTLSCipherRestrict.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

security.TestTLSCipherRestrict [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20498498?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20498498?buildTab=artifacts#/) on master @ [048cb14b78e76e6fbfb8292a044749d40acbfc5d](https://github.com/cockroachdb/cockroach/commits/048cb14b78e76e6fbfb8292a044749d40acbfc5d):


```
=== RUN   TestTLSCipherRestrict
    test_log_scope.go:171: test logs captured to: /artifacts/tmp/_tmp/7e9178a8651a0c2cb646d0ebdb5ed75e/logTestTLSCipherRestrict3656103733
    test_log_scope.go:82: use -show-logs to present logs inline
    tls_test.go:267: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/7e9178a8651a0c2cb646d0ebdb5ed75e/logTestTLSCipherRestrict3656103733
--- FAIL: TestTLSCipherRestrict (3.43s)
=== RUN   TestTLSCipherRestrict/invalid_ciphers
    test_server_shim.go:561: DRPC is enabled (override by TestingGlobalDRPCOption)
    tls_test.go:256: expected wantError=true, got err=<nil>
--- FAIL: TestTLSCipherRestrict/invalid_ciphers (0.86s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTLSCipherRestrict.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/security/BUILD.bazel
+++ b/pkg/security/BUILD.bazel
@@ -94,7 +94,6 @@ go_test(
         "//pkg/server",
         "//pkg/testutils",
         "//pkg/testutils/serverutils",
-        "//pkg/testutils/skip",
         "//pkg/util/envutil",
         "//pkg/util/leaktest",
         "//pkg/util/log",

--- a/pkg/security/tls_test.go
+++ b/pkg/security/tls_test.go
@@ -22,7 +22,6 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/security/username"
 	"github.com/cockroachdb/cockroach/pkg/testutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
-	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/cockroach/pkg/util/envutil"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
@@ -124,7 +123,6 @@ func verifyX509Cert(cert *x509.Certificate, dnsName string, roots *x509.CertPool
 func TestTLSCipherRestrict(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
-	skip.WithIssue(t, 153802)
 
 	tests := []struct {
 		name      string
@@ -164,7 +162,9 @@ func TestTLSCipherRestrict(t *testing.T) {
 			})()
 			ctx := context.Background()
 
-			s := serverutils.StartServerOnly(t, base.TestServerArgs{})
+			s := serverutils.StartServerOnly(t, base.TestServerArgs{
+				DefaultDRPCOption: base.TestDRPCDisabled,
+			})
 			defer s.Stopper().Stop(ctx)
 
 			// set the custom test ciphers

